src/charm.py

```diff
@@ -74,14 +74,12 @@ def _on_install(self, _):
         subprocess.check_output(["apt", "install", "-y", "apt-mirror"])
 
     def _on_config_changed(self, _):
-        for key in self.model.config:
-            if key not in self._stored.config:
-                value = self.model.config[key]
-                self._stored.config[key] = value
-            if self.model.config[key] != self._stored.config[key]:
-                value = self.model.config[key]
+        change_set = set()
+        for key, value in self.model.config.items():
+            if key not in self._stored.config or self._stored.config[key] != value:
                 logger.info("Setting {} to: {}".format(key, value))
                 self._stored.config[key] = value
+                change_set.add(key)
         if "use-proxy" in self._stored.config and self._stored.config["use-proxy"]:
```

Comment:
These lines(83-93) can be in separate function

Reply:
Done